stdlib/public/SwiftShims/LibcShims.h

```diff
@@ -104,14 +105,17 @@ static inline int _swift_stdlib_memcmp(const void *s1, const void *s2,
 
 // Non-standard extensions
 #if defined(__APPLE__)
+#define HAS_MALLOC_SIZE 1
 static inline __swift_size_t _swift_stdlib_malloc_size(const void *ptr) {
   extern __swift_size_t malloc_size(const void *);
   return malloc_size(ptr);
 }
 #elif defined(__linux__) || defined(__CYGWIN__) || defined(__ANDROID__) \
    || defined(__HAIKU__) || defined(__FreeBSD__) || defined(__wasi__)
+#define HAS_MALLOC_SIZE 1
 static inline __swift_size_t _swift_stdlib_malloc_size(const void *ptr) {
 #if defined(__ANDROID__)
+#define HAS_MALLOC_SIZE 1
```

Comment:
Does this `#define HAS_MALLOC_SIZE` twice for Android?

Reply:
Also, should `HAS_MALLOC_SIZE` be renamed to `_SWIFT_STDLIB_HAS_MALLOC_SIZE`?

Reply:
Good catch. Removed the Android #define.

I'm not sure a rename is necessary; giving it a full prefix suggests this is something with stronger scope than it actually has. The underscore prefix seems rather uncommon for stdlib as well, so I'm inclined to leave it as-is.

Reply:
Suggestion: replace each `#define HAS_MALLOC_SIZE` with a function?

```diff
-#define HAS_MALLOC_SIZE 1
+static inline __swift_bool _swift_stdlib_has_malloc_size() { return true; }
```

```diff
-#define HAS_MALLOC_SIZE 0
+static inline __swift_bool _swift_stdlib_has_malloc_size() { return false; }
```

Reply:
It's a lot more to copy and paste for the next new port, though.This suggests it would be far better to push the preprocessor conditionals into one definition of each function at the top level, but that would probably be better suited to do in a separate PR.